README.md

```diff
@@ -90,11 +87,32 @@ from root directory via `make tests` command, otherwise use standard PyTest synt
 https://universum.readthedocs.io/en/latest/configuring.html). Usage of such files
 is illustrated in `run_basic_example.sh` script.
 
-`analyzers` directory is not a part of Universum itself. It contains example external scripts
-compatible with Universum for static (and other types of) analysis.
+`analyzers` directory is not a part of Universum itself. It contains [example external scripts](
+https://universum.readthedocs.io/en/latest/code_report.html) compatible with Universum
+for static (and other types of) analysis.
+
+`setup.py` is 'setuptools' configuration file, and shouldn't be executed on its own.
 
+## Quick architecture overview
+
+The main idea of project 'Universum' architecture is ability to use it not only as
+finished product, but also as a library of separate and consistent modules.
 
-## Plan for improving README
+1. Project only entry point (except ['analyzers'](https://universum.readthedocs.io/en/latest/code_report.html))
+is `universum.py`. Based on chosen execution mode (default, submitting, polling, etc.)
+if calls one of 'main' modules, passing them all parameters
+2. Each module is inherited from `Module` class, defined in `gravity`. Because of this
```

Comment:
There is no proper transition from 1. to 2.
Please add another item and mention what are modules, that they are independent and other stuff like that.